megablocks/layers/moe.py

```diff
@@ -374,6 +375,9 @@ def parallel_forward_once(self, x, expert_weights, top_experts):
 
         # Locally permute the tokens and perform the expert computation.
         # Block to make sure that the cross-device permutation is complete.
+        if isinstance(self.mlp, mlp.GroupedMLP):
+            parallel_tokens_per_expert = parallel_tokens_per_expert_cpu.sum(
+                dim=0, dtype=torch.int)  # GroupedMLP requires counts on CPU
```

Comment:
"GroupedMLP requires counts on CPU"

The GroupedMLP will copy to CPU if the input isn't on CPU - can you add a comment explaining why it is desirable to avoid this copy? i.e., avoid an extra CPU-GPU synchronization point.